sdk/core/Azure.Core.TestFramework/src/ProxyTransport.cs

```diff
@@ -45,23 +45,49 @@ public ProxyTransport(TestProxy proxy, HttpPipelineTransport transport, TestReco
             _filter = filter;
         }
 
-        public override void Process(HttpMessage message)
-        {
-            RedirectToTestProxy(message);
-            _innerTransport.Process(message);
-            ProcessResponseAsync(message, false).EnsureCompleted();
-        }
+        public override void Process(HttpMessage message) =>
+            ProcessAsyncInternalAsync(message, false).EnsureCompleted();
 
-        public override async ValueTask ProcessAsync(HttpMessage message)
+        public override async ValueTask ProcessAsync(HttpMessage message) =>
+            await ProcessAsyncInternalAsync(message, true);
+
+        private async Task ProcessAsyncInternalAsync(HttpMessage message, bool async)
         {
-            RedirectToTestProxy(message);
-            await _innerTransport.ProcessAsync(message);
-            await ProcessResponseAsync(message, true);
+            try
+            {
+                RedirectToTestProxy(message);
+                if (async)
+                {
+                    await _innerTransport.ProcessAsync(message);
+                }
+                else
+                {
+                    _innerTransport.Process(message);
+                }
+
+                await ProcessResponseAsync(message, true);
+            }
```

Comment:
Should we add a `Catch` here just to log or to make debugging easier?

Reply:
The reason I added a finally is that some tests had policies that throw exceptions. I don't think there is any reason to catch these exceptions as they are part of the tests.